src/Uno.SampleGenerators/AdditionalFilesGenerator.cs

```diff
@@ -34,8 +34,13 @@ public void Execute(GeneratorExecutionContext context)
 			for (int i = 0; i < count; i++)
 			{
 				var additionalFile = context.AdditionalFiles[i];
+				if (!context.TryGetOptionValue(additionalFile, "build_metadata.AdditionalFiles.MyOption", out var myOptionValue))
+				{
+					myOptionValue = "Not found :(";
+				}
+
 				dictionaryStringBuilder.Append($@"
-							{{ ""{additionalFile.Path.Replace("\\", "\\\\")}"", ""{additionalFile.GetText()}"" }},
+							{{ ""{additionalFile.Path.Replace("\\", "\\\\")}"", ""{additionalFile.GetText()} -- MyOption: {myOptionValue}"" }},
```

Comment:
This is not working for some reason I can't figure out. :(

Reply:
@jeromelaban In `src\Uno.SampleProject\obj\Debug\netstandard2.0\Uno.SampleProject.GeneratedMSBuildEditorConfig.editorconfig`, I see the correct content:

```csharp
is_global = true
build_property.RootNamespace = Uno.SampleProject
build_property.ProjectDir = C:\Users\PC\Desktop\Uno.SourceGeneration\src\Uno.SampleProject\

[C:/Users/PC/Desktop/Uno.SourceGeneration/src/Uno.SampleProject/file1.txt]
build_metadata.AdditionalFiles.MyOption = This is file1.txt

[C:/Users/PC/Desktop/Uno.SourceGeneration/src/Uno.SampleProject/file2.txt]
build_metadata.AdditionalFiles.MyOption = This is file2.txt
```

However, this is the decompilation result of `Uno.SampleProject.dll`:

![image](https://user-images.githubusercontent.com/31348972/188941117-31d2ded0-7933-41d5-9ce7-48f80439da65.png)

Do you know if we are missing something [here](https://github.com/unoplatform/Uno.SourceGeneration/blob/d859a37defd43065b215a0d2dd3f68f6821e5064/src/Uno.SourceGeneration.Engine.Shared/Workspace/ProjectInfoBuilder.cs#L19)? or is the problem something else?

Reply:
This part comes from Roslyn, but I don't know where that could be missing. It could be specific to https://github.com/unoplatform/Uno.SourceGeneration/blob/d859a37defd43065b215a0d2dd3f68f6821e5064/src/Uno.SourceGeneration.Engine.Shared/Workspace/ProjectInfoBuilder.cs#L62.


But as I mentioned, if uno source gen does not support this mode, let's use AdditionalDocuments only in the netstandard2.0 in Uno.UI, and use standard files for net461.

Reply:
@jeromelaban I think I got it to work

![image](https://user-images.githubusercontent.com/31348972/188953741-74041ede-a429-4e7f-b602-c9e9cc8257e6.png)

Can you review the latest changes?